packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/GetAlbums.kt

```diff
@@ -0,0 +1,73 @@
+package expo.modules.medialibrary
+
+import android.content.Context
+import android.os.AsyncTask
+import android.os.Bundle
+import android.provider.MediaStore
+import android.provider.MediaStore.Images.Media
+import org.unimodules.core.Promise
+
+internal open class GetAlbums(private val mContext: Context, private val mPromise: Promise)
+  : AsyncTask<Void?, Void?, Void?>() {
+
+  override fun doInBackground(vararg params: Void?): Void? {
+    val projection = arrayOf(Media.BUCKET_ID, Media.BUCKET_DISPLAY_NAME)
+    val selection = "${MediaStore.Files.FileColumns.MEDIA_TYPE} != ${MediaStore.Files.FileColumns.MEDIA_TYPE_NONE}"
+
+    val albums = HashMap<String, Album>()
+
+    try {
+      mContext.contentResolver
+        .query(
+          MediaLibraryConstants.EXTERNAL_CONTENT,
+          projection,
+          selection,
+          null,
+          Media.BUCKET_DISPLAY_NAME)
+        .use { asset ->
+          if (asset == null) {
+            mPromise.reject(MediaLibraryConstants.ERROR_UNABLE_TO_LOAD,
+              "Could not get albums. Query returns null.")
+            return@use
+          }
+          val bucketIdIndex = asset.getColumnIndex(Media.BUCKET_ID)
+          val bucketDisplayNameIndex = asset.getColumnIndex(Media.BUCKET_DISPLAY_NAME)
+
+          while (asset.moveToNext()) {
+            val id = asset.getString(bucketIdIndex)
+
+            val album = albums[id] ?: let {
+              val newAlbum = Album(
+                id = id,
+                title = asset.getString(bucketDisplayNameIndex)
+              )
+              albums[id] = newAlbum
+
+              newAlbum
+            }
```

Comment:
```suggestion
            val album = albums[id] ?: Album(
              id = id, 
              title = asset.getString(bucketDisplayNameIndex)
             ).also {
            	 albums[id] = it
             }
```
I'm not sure how the correct formatting should look like here 😅

Reply:
It can be even nicer ;) Formatting looks ok, possibly _also_ block could be inlined.